transcom/mymove
# B-17748 TOO SIT Entry Date Edit with Correct Service Codes and Audit Log

## [Update Current SIT to Use Correct Service Codes](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A854098&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

This story focuses on using the appropriate service codes when editing a SIT entry date. It was found previously the service codes `DOPSIT` and `DDDSIT` were being used when considering the SIT entry date. This was incorrect and needed to be changed to `DOFSIT` (Origin 1st day SIT) and `DDFSIT` (Destination 1st day SIT).

In addition to this change, those service items have "sister" service items of `DOFSIT` and `DDASIT`, respectively, that deal with the add'l days of SIT. Those service items need to have a SIT entry date of the NEXT day since they cannot be the same day as the 1st day of SIT and are used later on to calculate values in the `payment requests`.

We also want to be able to track these changes in the database, so functionality was added to show details in the audit logs as well as adding the `ToolTip` feature to show the previous values prior to updating the SIT entry date.

If a DOFSIT or DDFSIT item has a payment request created then the Edit button will be disabled for the TOO and a red ToolTip will display and inform the TOO & TIO that the SIT Entry Date cannot be changed for that item because a payment request exists for it.

Some minor changes in this code as well to fix some previous issues:
- validation on the SIT entry date form to not allow submission if it is the same date as the previous SIT entry date as well as setting the `touched` value to `false` in Formik that disables the Save button until the SIT entry date calendar button is clicked as well as office remarks being filled out
- Removed error data from the `SomethingWentWrong` component
- Had to add `MockProviders` to a lot of existing tests in order to pass the `moveCode` parameter since it is being used in the `ServiceItemsTable` component

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the Office MilMove application
2. Login as a TOO
3. Update the SIT entry date of a service item at the bottom of the `Move Task Order` screen
4. Following update, you should see a green banner, the SIT panel and the service item updated, as well as logs added to the `Move History` tab<br><br>THEN<br><br>
1. Access the Office MilMove application
2. Login as any other role except TOO (do not use the multi-role option)
3. Make sure you cannot edit the SIT entry date of any service item<br><br>THEN<br><br>
1. Access the Office MilMove application<br>
2. Login as a TOO<br>
3. Find DOFSIT or DDFSIT service item that has an existing payment request and verify that the Edit button is disabled and a ToolTip displays
